### PR TITLE
Pscoast longopts

### LINKED ISSUE #6905: Add long-options to pscoast
**Description of the desired feature**

Add long-options to pscoast.

**Are you willing to help implement and maintain this feature?** Yes/No

Yes

### PR DESCRIPTION
**Pull request by rbdavis**
**Description of proposed changes**

Add long-options support in pscoast

Fixes #6905 

**Reminders**

- [ ] Make sure that your code follows our style. Use the other functions/files as a basis.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Describe changes to function behavior and arguments in a comment below the function declaration.
- [ ] If adding new functionality, add a detailed description to the documentation and/or an example.

### CODE REVIEW

File: src/gmt_init.c [hunk 1]
```diff
@@ -518,50 +522,80 @@ GMT_LOCAL void gmtinit_handle_escape_text (char *text, char key, int way) {
...
 
+/*! . */
+GMT_LOCAL char * gmtinit_strchr_predexcl (char *string, char target, char predexcl) {
+	/* Version of strchr() that will ignore any instance of target
+	 * that is preceded by predexcl, the predecessor character
+	 * whose presence will exclude that instance from selection. */
+	char *c;
+	if (string == NULL) return NULL;
+	for (c = string; *c != '\0'; c++)
+		if ((*c == target) &&
```
Comment (joa-quim) [inline src/gmt_init.c:533]: With our code style these two lines would be joined in one. We don't privilege the "vertical" styling.

Comment (rbdavis) [reply src/gmt_init.c:533]: Which two lines exactly?

Comment (PaulWessel) [reply src/gmt_init.c:533]: I think @joa-quim means that the if-test on 533 should continue on that ine and not break at the &&, right?

Comment (joa-quim) [reply src/gmt_init.c:533]: Yes, that's where I inserted the comment.

Comment (rbdavis) [reply src/gmt_init.c:533]: OK. All right with all if I fix this in the next branch which I should be pushing later today? I'd prefer not to mess around in multiple branches on my local system, as I'm currently working on the next batch of updates for pswiggle (where I will incorporate this change).

Comment (joa-quim) [reply src/gmt_init.c:533]: Sure, no problems. And you can even do it directly from the GH page.

Comment (rbdavis) [reply src/gmt_init.c:533]: All right, now I see what you meant when you said that's where you inserted the comment. Still not clear to me how to change the code from that page so I'll just leave it alone until the next branch is done.

Comment (joa-quim) [reply src/gmt_init.c:533]: Simple,
image

Comment (rbdavis) [reply src/gmt_init.c:533]: Thanks for the tip! (But will still wait for the next branch update.)

Comment (seisman) [reply src/gmt_init.c:533]: Another tip here: you can just type `.` on this page, and you'll be redirected to the github.dev site, in which you can edit any files using the web-based vscode editor. I think it's the best editing experience using a web browser.

See https://github.com/github/dev for the official documentation.

Comment (rbdavis) [reply src/gmt_init.c:533]: Thanks for that info!

**Review by PaulWessel — APPROVED** [review]
Good to go!

**Pull request merged**